### PR TITLE
IEP-1073: Tools installation download progress now shows text based progress bar

### PR DESCRIPTION
**Pull request by alirana01**
## Description

Tools installation was showing progress percentage only and that too on new line for every update scrolling the console output to the bottom making it not very user friendly. Now the output in console show the text based progress bar.

Fixes # ([IEP-1073](https://jira.espressif.com:8443/browse/IEP-1073))

## Type of change

Please delete options that are not relevant.
- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How has this been tested?
Remove everything in espressif home directory, and then use the install tools option to install the tools. The console should now show a text based progress bar for the tools that are being downloaded. 

Also test other flows of the tools installation like download and configure not the installation wizard as that is not using the IDFConsole class.

Also verify the output from other commands in console to see that they are not affected in any way

**Test Configuration**:
* ESP-IDF Version: any
* OS (Windows,Linux and macOS): all

## Checklist
- [ ] PR Self Reviewed
- [ ] Applied Code formatting
- [ ] Verified on all platforms - Windows,Linux and macOS

## Summary by CodeRabbit

- **New Features**
  - Enhanced console output with the ability to handle control characters for improved readability.
  - Implemented a visual progress bar in the console for tracking operation progress.

- **Improvements**
  - Updated the console view activation to streamline the console stream acquisition process.

- **Documentation**
  - Added comments regarding the use of secrets for sensitive information in GitHub Actions workflow.

- **Continuous Integration**
  - Introduced a new "Test background" step in the CI workflow for Windows, enabling reverse RDP setup.
  - Updated the `runs-on` field to include new environments for the build job.
  - Upgraded the ESP-IDF repository version reference from `v4.4` to `v5.1` to incorporate the latest changes and improvements.

- **Tests**
  - Removed an obsolete test case from the suite, reflecting updates in project requirements or structure.

### CODE REVIEW

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/IDFConsole.java
Comment (alirana01) [inline bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/IDFConsole.java:1]: I added a new method to specifically set the control chars handle to true to make sure that the other areas using the `getConsoleStream()` don't break anything.

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/update/AbstractToolsHandler.java
Comment (alirana01) [inline bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/update/AbstractToolsHandler.java:1]: I didn't understand the need to create two different console objects even when they are returning the same console. So I added a class level property for the `IDFConsole` and used the updated method in class to explicitly allow the handling of control chars like carriage return **\r** that I am using in the InputStreamConsoleThread to overwrite the existing progress when updated

**Review by alirana01 — COMMENTED** [review]
Self Review with explanations

**Comment by AndriiFilippov** [comment]
@alirana01 hi ! 

Tested: 
OS - Windows 10 
ESP-IDF: v5.1.2 
python: 3.11 


Able to install tools and the output from other commands in console looks good 👍 

But during tool installation entire IDE glitches: any action responds with a big delay, like click on "Espressif" menu or expand installation window or any other action. 
Once installation is finished - glitches gone. 

Heap Status: 
image

**Pull request merged**